pkg/resource/plugin/provider.go

```diff
@@ -71,6 +72,11 @@ type Provider interface {
 	Delete(urn resource.URN, id resource.ID, props resource.PropertyMap, timeout float64) (resource.Status, error)
 	// Invoke dynamically executes a built-in function in the provider.
 	Invoke(tok tokens.ModuleMember, args resource.PropertyMap) (resource.PropertyMap, []CheckFailure, error)
+	// StreamInvoke dynamically executes a built-in function in the provider, which returns a stream
+	// of responses.
+	StreamInvoke(
+		tok tokens.ModuleMember, args resource.PropertyMap,
+		stream pulumirpc.ResourceMonitor_StreamInvokeServer) (resource.PropertyMap, []CheckFailure, error)
```

Comment:
This doesn't feel like the "right" type signature. Can we abstract out the `StreamInvokeServer` bit? I'm also surprised that this returns anything at all given the implementation in the `QueryResourceMonitor`.

Reply:
Yes.